gammapy/data/gti.py

```diff
@@ -438,6 +438,56 @@ def union(self, overlap_ok=True, merge_equal=True):
         merged = Table(rows=merged, names=["START", "STOP"], meta=self.table.meta)
         return self.__class__(merged, reference_time=self.time_ref)
 
+    def intersect(self, other, padding=0.0):
+        """Intersection of overlapping time intervals between two GTIs.
+
+        Returns a new `~gammapy.data.GTI` object.
+
+        Parameters
+        ----------
+        other : `~gammapy.data.GTI`
+            Another GTI object to intersect with.
+        padding : float, optional
+            Padding in seconds to expand the intersection intervals. Default is 0.0.
+
+        Returns
+        -------
+        intersected_gti : `~gammapy.data.GTI`
+            A GTI object whose intervals are the intersections of the two GTIs.
+        """
+        from astropy.time import TimeDelta
+
+        intervals = []
+        padding_delta = TimeDelta(padding, format="sec")
```

Comment:
why a padding is this needed ?
don't you have to ensure that the padding do not add overlap between the output intervals ?